src/swarm-tui/Swarm/TUI/View.hs

```diff
@@ -232,14 +235,19 @@ drawNewGameMenuUI (l :| ls) launchOptions = case displayedFor of
     (Nothing, Just (SISingle _)) -> hCenter $ txt "Press 'o' for launch options, or 'Enter' to launch with defaults"
     _ -> txt " "
 
-  drawScenarioItem (SISingle (s, si)) = padRight (Pad 1) (drawStatusInfo s si) <+> txt (s ^. scenarioMetadata . scenarioName)
+  drawScenarioItem (SISingle (s, ScenarioPath sPath)) = padRight (Pad 1) (drawStatusInfo s sPath) <+> txt (s ^. scenarioMetadata . scenarioName)
   drawScenarioItem (SICollection nm _) = padRight (Pad 1) (withAttr boldAttr $ txt " > ") <+> txt nm
-  drawStatusInfo s si = case si ^. scenarioStatus of
-    NotStarted -> txt " ○ "
-    Played _script _latestMetric best | isCompleted best -> withAttr greenAttr $ txt " ● "
-    Played {} -> case s ^. scenarioOperation . scenarioObjectives of
-      [] -> withAttr cyanAttr $ txt " ◉ "
-      _ -> withAttr yellowAttr $ txt " ◎ "
+
+  drawStatusInfo s sPath = case currentScenarioInfo of
+    Nothing -> emptyWidget
+    Just si -> case si ^. scenarioStatus of
+      NotStarted -> txt " ○ "
+      Played _script _latestMetric best | isCompleted best -> withAttr greenAttr $ txt " ● "
+      Played {} -> case s ^. scenarioOperation . scenarioObjectives of
+        [] -> withAttr cyanAttr $ txt " ◉ "
+        _ -> withAttr yellowAttr $ txt " ◎ "
+   where
+    currentScenarioInfo = appState ^? runtimeState . scenarios . scenarioItemByPath sPath . _SISingle . _2
```

Comment:
The status indicator will not be rendered if the path cannot be looked up in the toplevel `scenarios` collection.